multiqc/multiqc.py

```diff
@@ -746,6 +746,27 @@ def __rich_measure__(self, console: rich.console.Console, options: rich.console.
         report.runtimes["mods"][run_module_names[mod_idx]] = time.time() - mod_starttime
     report.runtimes["total_mods"] = time.time() - total_mods_starttime
 
+    # Parse software version from config if provided
+    versions_from_config = software_versions.load_versions_from_config(config)
+    if versions_from_config:
+        for software, versions in versions_from_config.items():
+            # Try to find if the software is listed among the executed modules.
+            # Unlisted software are still reported in the `Software Versions` section.
+            versions = list(set(map(str, versions)))
+            versions.sort(reverse=True)
+            module = software_versions.find_matching_module(software, report.modules_output)
+            if module is None:
+                logger.debug(f"No executed modules matches '{software}' listed in config.software_versions")
+            else:
+                software = module.name
+                for version in versions:
+                    module.add_software_version(str(version))
+            report.software_versions[software] = versions
+
+    # Add section for software versions if any are found
+    if report.software_versions and len(report.modules_output) > 0:
```

Comment:
We need a new config option to disable this, people may not always want it in reports.

Reply:
Added a new option `--no-versions-section` for this.

Reply:
Ok cool - I didn't really mean CLI flag, I just meant something that can be set in a config file. I think I'd prefer to remove the new CLI flag if you don't mind. Sorry for not being clear.